packages/vite/src/node/plugin.ts

```diff
@@ -36,8 +43,132 @@ import type { PreviewServerHook } from './preview'
  *
  * If a plugin should be applied only for server or build, a function format
  * config file can be used to conditional determine the plugins to use.
+ *
+ * The current module environment can be accessed from the context for the
+ * buildStart, resolveId, transform, load, and buildEnd, hooks
+ *
+ * The current environment can be accessed from the context for the
+ * buildStart, resolveId, transform, load, and buildEnd, hooks. It can be a dev
+ * or a build environment. Plugins can use this.environment.mode === 'dev' to
+ * check if they have access to dev specific APIs.
```

Comment:
What is the difference between "current module environment" and "current environment"?

Reply:
The first two lines are from an outdated refactoring that ended up duplicating the comment. Removed 👍🏼